docs/fundamentals/app-lifecycle.md

```diff
@@ -0,0 +1,440 @@
+| `Deactivated` | This event is raised when the window is no longer the focused window. However, the window might still be visible. |  |
+| `Stopped` | This event is raised when the window is no longer visible. There's no guarantee that an app will resume from this state, because it may be terminated by the operating system. | Disconnect from any long running processes, or cancel any pending requests that might consume device resources. |
+| `Resumed` | This event is raised when an app resumes after being stopped. This event won't be raised the first time your app launches, and can only be raised if the `Stopped` event has previously been raised. | Subscribe to any required events, and refresh any content that's on the visible page. |
+| `Destroying` | This event is raised when the native window is being destroyed and deallocated. The same cross-platform window might be used against a new native window when the app is reopened. | Remove any event subscriptions that you've attached to the native window. |
+
+These cross-platform events map to different platform events, and the following table shows this mapping:
+
+| Event | Android | iOS | Windows |
+| -- | -- | -- | -- |
+| `Created` | `OnPostCreate` | `FinishedLaunching` | `Created` |
+| `Activated` | `OnResume` | `OnActivated` | `Activated` (`CodeActivated` and `PointerActivated`) |
+| `Deactivated` | `OnPause` | `OnResignActivation` | `Activated` (`Deactivated`) |
+| `Stopped` | `OnStop` | `DidEnterBackground` | `VisibilityChanged` |
+| `Resumed` | `OnRestart` | `WillEnterForeground` | `Resumed` |
+| `Destroying` | `OnDestroy` | `WillTerminate` | `Closed` |
+
+In addition to these events, the `Window` class also has the following overridable methods:
+
+- `OnCreated`, which is invoked when the `Created` event is raised.
+- `OnActivated`, which is invoked when the `Activated` event is raised.
+- `OnDeactivated`, which is invoked when the `OnDeactivated` event is raised.
+- `OnStopped`, which is invoked when the `Stopped` event is raised.
+- `OnResumed`, which is invoked when the `Resumed` event is raised.
+- `OnDestroying`, which is invoked when the `Destroying` event is raised.
+
+To subscribe to the `Window` lifecycle events, override the `CreateWindow` method in your `App` class to create a `Window` instance on which you can subscribe to events:
+
+```csharp
+using Microsoft.Maui;
+using Microsoft.Maui.Controls;
+using Application = Microsoft.Maui.Controls.Application;
+
+namespace MyMauiApp
+{
+    public partial class App : Application
+    {
+        public App()
+        {
+            InitializeComponent();
+
+            MainPage = new MainPage();
+        }
+
+        protected override Window CreateWindow(IActivationState activationState)
+        {
+            Window window = base.CreateWindow(activationState);
+
+            window.Created += (s, e) =>
+            {
+                // Register services
```

Comment:
I don't think this is correct. Services should all be registered inside the builder. Unless I'm not following what you're saying here.